### PR TITLE
(PDK-1269) Add `pdk new plan`

### PR DESCRIPTION
**Pull request by npwalker**
This is a copy / paste of `pdk new task` to get a start on `pdk new plan` 

Requires https://github.com/puppetlabs/pdk-templates/pull/184

### CODE REVIEW

File: lib/pdk/validate/plans/metadata_lint.rb [hunk 1]
```diff
@@ -0,0 +1,126 @@
+require 'pdk'
```
Comment (bmjen) [inline lib/pdk/validate/plans/metadata_lint.rb]: This file to be removed, there's no metadata file for plans.

File: lib/pdk/cli/util/option_validator.rb [hunk 2]
```diff
@@ -23,6 +23,7 @@ def self.valid_module_name?(string)
           !(string =~ %r{\A[a-z][a-z0-9_]*\Z}).nil?
         end
         singleton_class.send(:alias_method, :valid_task_name?, :valid_module_name?)
+        singleton_class.send(:alias_method, :valid_plan_name?, :valid_module_name?)
```
Comment (bmjen) [inline lib/pdk/cli/util/option_validator.rb]: Plans also require checking the name against the list of reserved words, will need to account for that somehow.

**Comment by geoffnichols** [comment]
@npwalker I noticed this hasn't seen any activity in a few months, do you think you'll be able to work on it in the near future?

**Comment by geoffnichols** [comment]
Since this hasn't seen any activity in about 6 months I'm going to go ahead and close this pull request for the time being. Please re-open this pull request once the next actions are addressed, new information is available, or you have a question related to this pull request.

**Pull request closed without merging**